blocks navigation + import/export + menu highlight

TaunT opened the pull request:
In manual mode:
1.  Highlighting the top menu during work
![image](https://github.com/user-attachments/assets/fd6cd072-b43f-415d-b1a6-5608d1fb1a46)

2. Navigation between text blocks
3. Import and export of translations
![image](https://github.com/user-attachments/assets/3e3f4e4a-1439-4abb-b9eb-47540a3b6ab7)

TaunT commented:
@ogkalu2 Thank you for your trust and the great honor of participating in the project =)
What about this functionality, do you not like the meaning of the additions, or are there problems with the code?

TaunT commented:
@ogkalu2 When merging a branch with text size and color, 3 lines were lost, or did you remove them? Block font settings buttons are inactive in automatic mode

**comic.py**
```
    def batch_mode_selected(self):
        ...
        self.set_font_settings_enabled(False)

    def manual_mode_selected(self):
        ...
        self.set_font_settings_enabled(True)
```

**main_window.py**
```
    def set_font_settings_enabled(self, enabled: bool):
        self.min_font_spinbox.setEnabled(enabled)
        self.max_font_spinbox.setEnabled(enabled)
        self.block_font_color_button.setEnabled(enabled)
```

ogkalu2 commented:
> @ogkalu2 When merging a branch with text size and color, 3 lines were lost, or did you remove them? Block font settings buttons are inactive in automatic mode
> 
> **comic.py**
> 
> ```
>     def batch_mode_selected(self):
>         ...
>         self.set_font_settings_enabled(False)
> 
>     def manual_mode_selected(self):
>         ...
>         self.set_font_settings_enabled(True)
> ```
> 
> **main_window.py**
> 
> ```
>     def set_font_settings_enabled(self, enabled: bool):
>         self.min_font_spinbox.setEnabled(enabled)
>         self.max_font_spinbox.setEnabled(enabled)
>         self.block_font_color_button.setEnabled(enabled)
> ```

Not sure what happened there but i've added it back. i didn't delete it.

ogkalu2 commented:
> @ogkalu2 Thank you for your trust and the great honor of participating in the project =) What about this functionality, do you not like the meaning of the additions, or are there problems with the code?
> 
> 1. I use import and export very often myself. And I added the coordinates of the blocks there, now they are also saved and loaded - now it is very easy to re-edit the page.
> 2. During the translation process, I sometimes switch to other tasks or get distracted by looking at photos of cute cats or boobs, so before I made the menu highlighting, I sometimes made a mistake about which step I was on and clicked the wrong button
> 3. It is convenient to scroll through the blocks to check all the translations for errors again
> 
> p.s. rebased to fresh main branch

1. Saving the state of the app as a Project (https://github.com/ogkalu2/comic-translate/issues/101) is a better idea i think. I don't want to clutter the main window before that with buttons just to save text. 
2. The app isn't really strictly meant to be used in order. Someone might skip or interchange between the buttons. Can you implement the highlighting such that it only highlights the last used button ? 
3. I don't mind this at all.

TaunT commented:
@ogkalu2 "Saving the state of the app as a Project (https://github.com/ogkalu2/comic-translate/issues/101)"
In fact, my export-import now saves the necessary information to load it later. But only for one picture in manual mode. It can be improved, and saved for all, and when importing, draw all the blocks again (now the coordinates of the current ones are simply changed) I have enough of this in my work for now, then I will try to finish it when I have time, if you don’t do it earlier.

TaunT commented:
@ogkalu2 Hi! I improved my import-export, it seems to me that it can be conveniently worked with manually until you make a general save of the project. Maybe take a look? Is it possible to merge this into the project?
![image](https://github.com/user-attachments/assets/fb5df839-8dd1-4f5b-9397-d0ada0551672)

TaunT commented:
@ogkalu2 Thanks for the fix for drawing and deleting rectangles!
I rebased this branch and cleaned it up, here is ONLY the new manual mode panel with export and import of finished translation and clean image.
(Maybe it will be easier for you to decide to add this to the project)

Now it was possible to load the cleaned image, blocks and easily change them a little.

notimp commented:
As I reported in the other pull request, I sometimes get recctangles that lose the text thats attached to them, when I do a final control pass on "automatic mode" runs. About twice every comic.

If "save state" could be implemented in a way where it would not be "overwritten" automatically. Or if a "save as (with a different filename)" for save state would be possible, that would help me with those cases - as as soon as a rectange loses its attached (auto translated) text, I could reload the state to get it back.

Its a work around, but if you could do that on a "per page basis" (can still be one project file), that would be a workable workaround for me loosing text thats attached to rectangles in the quality control pas after an automatic mode run. Currently, as soon as that happens, I have to stop working on the image, and continue with the next one, so I dont loose the auto recognized text on the image.

After that the image stays as auto mode saved it, because I have no way of getting the recognized states back into comic-translate currently.

Thank you.

notimp commented:
Also by default probably save those state files within the translation folder of the project. So you dont run into issues where the state files get loaded for different projects, because the file names were the same. (001.jpg, 002.jpg, and so on.)

Just as a reminder in case you thought about storing them centrally. ;)

TaunT commented:
@notimp of course a date and time stamp is added so as not to accidentally overwrite it)
```
RCO002.jpg
RCO002_2024-08-24_13-33-21_cleaned.jpg
RCO002_2024-08-24_13-39-34.txt
```

TaunT commented:
@ogkalu2 rebased on current master
